### PR TITLE
Fix a bunch of command/alias issues

### LINKED ISSUE #366: Submodules ModuleInfo.Aliases collection does not contain Group name
```cs
[Group("primary")]
public class PrimaryModule : ModuleBase { 
    [Group("sub")]
    public class SubModule : ModuleBase {}
}
```
"primary" will be found in PrimaryModule's aliases, while "sub" will not be in SubModule's aliases (collection will be empty in this case). The behavior only happens to submodules.

### PR DESCRIPTION
**Pull request by FiniteReality**
According to my testing, resolves #366, #365 as well as CommandService.Commands/Modules not returning all registered commands/modules respectively.

### CODE REVIEW

File: src/Discord.Net.Commands/Builders/ModuleClassBuilder.cs [hunk 1]
```diff
@@ -127,7 +129,9 @@ private static void BuildCommand(CommandBuilder builder, TypeInfo typeInfo, Meth
                     builder.Name = builder.Name ?? cmdAttr.Text;
                 }
                 else if (attribute is NameAttribute)
+                {
```
Comment (RogueException) [inline src/Discord.Net.Commands/Builders/ModuleClassBuilder.cs]: Style consistency: remove braces

File: src/Discord.Net.Commands/Builders/ModuleClassBuilder.cs [hunk 2]
```diff
@@ -140,6 +144,9 @@ private static void BuildCommand(CommandBuilder builder, TypeInfo typeInfo, Meth
                     builder.AddPrecondition(attribute as PreconditionAttribute);
             }
 
+            if (builder.Name == null)
+                builder.Name = method.Name;
```
Comment (RogueException) [inline src/Discord.Net.Commands/Builders/ModuleClassBuilder.cs]: We already have this on line 119

**Review by Auralytical — CHANGES_REQUESTED** [review]

**Comment by james7132** [comment]
Just tested this on my current setup, and it does not resolve #365: there is still a null alias included. Will report back why as soon as possible.

**Comment by james7132** [comment]
The following code still triggers a NullReferenceException as described in #365:

```csharp
public class Admin : ModuleBase {

  [Group("prune")]
  public class Prune : ModuleBase {

    [Command]
    public Task Default(int count) {
      ...
    }

  }

}
```

The builder has a single null alias and the module has no aliases in this case.

**Comment by FiniteReality** [comment]
@james7132 Interesting; it appears to be fixed for me:
screenshot_20161125_204315

Mind showing your code?

**Comment by james7132** [comment]
For the exact code, check [here](https://github.com/james7132/Hourai/blob/master/src/Modules/Admin/Prune.cs#L18).

**Comment by james7132** [comment]
Yes I am currently on commit 05fb81c.

**Comment by FiniteReality** [comment]
@foxbot I can reproduce the issue; my Permutate code is returning an empty result which is incorrect

**Comment by FiniteReality** [comment]
Okay, it should be fixed now. I'll work on fixing the reviewed issues now.

**Comment by james7132** [comment]
Can confirm that this resolves  #365 now.

**Review by Auralytical — APPROVED** [review]

**Pull request merged**